### PR TITLE
[Users] Import a list of users by copy/paste from Excel file

### PR DESCRIPTION
**Pull request by ciprianbolocan**

### CODE REVIEW

File: src/main/java/com/code4ro/legalconsultation/service/impl/UserService.java [hunk 1]
```diff
@@ -127,4 +125,30 @@ public void deleteById(final String id) {
...
+
+    public List<UserDto> extractFromCopyPaste(List<String> usersListAsString) {
+        List<User> usersList = new ArrayList<>();
+        List<String[]> split = usersListAsString.stream()
+                .map(user -> user.split(COMMA_REGEX))
+                .collect(Collectors.toList());
+
+        split.forEach( userProperty ->
+                usersList.add(
+                        new User(
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/service/impl/UserService.java]: why not create a `UserDto` instead?

**Comment by ciprianbolocan** [comment]
I thought about this, but in the dto there was no constructor, and i thought that should be a reason for that. It’s ok to add a builder to UserDto and to create one ?

**Comment by mihairinzis** [comment]
> I thought about this, but in the dto there was no constructor, and i thought that should be a reason for that. It’s ok to add a builder to UserDto and to create one ?

yes, maybe use the `@Builder` annotation from lombok

File: src/main/java/com/code4ro/legalconsultation/service/impl/UserService.java [hunk 2]
```diff
@@ -127,4 +125,30 @@ public void deleteById(final String id) {
         final ObjectReader reader = csvMapper.readerFor(User.class).with(schema);
         return reader.<User>readValues(stream).readAll();
     }
+
+    public List<UserDto> extractFromCopyPaste(List<String> usersListAsString) {
+        List<User> usersList = new ArrayList<>();
+        List<String[]> split = usersListAsString.stream()
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/service/impl/UserService.java]: would it be possible to convert thie user list to an input stream and reuse the `read(InputStream stream)` method?

**Comment by ciprianbolocan** [comment]
Ok, I will modify it today to reuse read method and commit with ammend today or tomorrow

**Comment by ciprianbolocan** [comment]
yep, better this approach. thanks for it ^^

File: src/main/java/com/code4ro/legalconsultation/model/persistence/User.java [hunk 3]
```diff
@@ -54,6 +51,18 @@ public User(final String email, final UserRole role) {
         this.role = role;
     }
 
+    public User(String firstName, String lastName, String email,
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/model/persistence/User.java]: don't think you need this anymore :)

File: src/main/java/com/code4ro/legalconsultation/model/dto/UserDto.java [hunk 4]
```diff
@@ -28,4 +28,6 @@
     private String organisation;
     @NotNull
     private UserRole role;
+
+    public UserDto(){ }
```
Comment (RaduCStefanescu) [inline src/main/java/com/code4ro/legalconsultation/model/dto/UserDto.java]: Do you need this empty constructor? Isn't it default?

File: src/main/java/com/code4ro/legalconsultation/model/persistence/User.java (hunk 3 above)
Comment (RaduCStefanescu) [reply src/main/java/com/code4ro/legalconsultation/model/persistence/User.java]: Yeah, good question. Do you use this constructor anywhere?

**Comment by RaduCStefanescu** [comment]
hei @ciprianbolocan thank you for the PR
just some small changes and we should be able to merge
checkout the comments

File: src/main/java/com/code4ro/legalconsultation/model/persistence/User.java (hunk 3 above)
Comment (ciprianbolocan) [reply src/main/java/com/code4ro/legalconsultation/model/persistence/User.java]: yep, don't need this anymore, i'll delete it

File: src/main/java/com/code4ro/legalconsultation/model/dto/UserDto.java (hunk 4 above)
Comment (ciprianbolocan) [reply src/main/java/com/code4ro/legalconsultation/model/dto/UserDto.java]: Yes, the mapper service is throwing an exception if i don't have the default constructor. That's why I put it. You can see also that the User class has an empty constructor.

**Pull request merged**